adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -698,6 +703,81 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
   }
 
   /**
+   * This method gets the sort and partition map metadata from the header of the file
+   * give as input.
+   *
+   * @param filename the filename for the metadata
+   * @return a partition map if the data was written sorted, or an empty Seq if unsorted
+   */
+  private[rdd] def extractPartitionMap(filename: String): Seq[(ReferenceRegion, ReferenceRegion)] = {
+    // the sorted metadata is always stored with the sequence dictionary metadata
+    val path = new Path(filename + "/_seqdict.avro")
+    val fs = path.getFileSystem(sc.hadoopConfiguration)
+
+    try {
+      // get an input stream
+      val is = fs.open(path).asInstanceOf[InputStream]
+
+      // set up avro for reading
+      val dr = new GenericDatumReader[GenericRecord]
+      val fr = new DataFileStream[GenericRecord](is, dr)
+
+      // parsing the json from the metadata header
+      // this unfortunately seems to be the only way to do this
+      // avro does not seem to support getting metadata fields out once
+      // you have the input from the string
+      val metaDataMap = JSON.parseFull(fr.getMetaString("avro.schema"))
```

Comment:
@heuermh The current implementation is mapped out as follows:

1.) We create a new metadata file for reading in the partition map (see line 758 of ADAMContext.scala)
2.) We read/write the data as JSON.